keps/sig-network/2079-network-policy-port-range/README.md

```diff
@@ -0,0 +1,623 @@
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+In NetworkPolicy specification, inside ``NetworkPolicyPort`` object struct specify a new ``PortRange`` field:
+
+```
+// NetworkPolicyPort describes a port to allow traffic on
+type NetworkPolicyPort struct {
+    PortRange *string
+}
+```
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As the owner of a web application I want to allow the traffic to my Pods in different ports without having to create multiple network policies. As an example, my application is exposed in ports 80, 443 and 8443 and I want to group this in a single rule
+
+#### Story 2
+
+I have another application that communicates with NodePorts of a different cluster and I want to allow the egress of the traffic only the NodePort range (eg. 30000-32767) as I don't know which port is going to be allocated in the other side, but don't want to create a rule for each of them.
+
+#### Story 3
+
+As the owner of an application, I want to allow the communication with a set of ports between 6000 and 7000 except port 6667.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+CNIs will need to support the new field in their controllers. For this case we'll try to make broader communication with the main CNIs so they can be aware of the new field.
+
+The field will need some syntax validation either, to not allow the mix of named and non-named ports inside a range (like http-9000), and also to allow only numbered ports inside uint16 range.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+API changes to NetworkPolicy:
+* Add a new field ``spec.ingress|egress.ports.PortRange
```

Comment:
Wondering if it makes sense to have `portRange` under `ports` or should `ports` and `portRanges` be separate fields under `ingress` and `egress`? What happens if `portRanges` conflicts with `port`?

Reply:
I think it's up to the CNI how it's going to deal with the conflicts? Or should we validate this in the API?

 But I have no opposition to move portRanges to outside ports, whatever is cleaner/better

Reply:
if we move port under portRange(or maybe some new name, so having port under portRange doesn't sound awkward) and validate at API level that only one of those fields is set, we can eliminate potential conflicts

Reply:
do we want to eliminate the conflict, or consider only one field? 

Or maybe convert a .port entry to a .portRange.port as proposed in the comment below

Also, about the naming thing, maybe .ports.Range can be a better naming?

Reply:
I think it would be useful to see all the example yaml for all the options, maybe in a separate google doc for now?